### PR TITLE
Check if service field is provided by auth challenge

### PR DESCRIPTION
**Pull request by jiayihu**
This solves #469 It should be a no brainer.

Trying to request the image from Google Artifacts returns `Bearer realm="https://europe-west1-docker.pkg.dev/v2/token"` without the `service` field. The [Distribution Spec](https://github.com/opencontainers/distribution-spec/blob/master/spec.md) doesn't tell anything about authentication AFAIK.

I found about the service field in the [Docker Registry docs](https://docs.docker.com/registry/configuration/#silly), so it's probably not needed anymore in Artifacts?

### CODE REVIEW

**Comment by bacongobbler** [comment]
Hi @jiayihu! Would you mind writing a unit test to cover this test case? That way we can ensure this scenario is tested and accounted for in future iterations. Thanks!

**Comment by jiayihu** [comment]
Sure, but I think that would need me to add an image url here: 

https://github.com/deislabs/krustlet/blob/592bb8d20cf729fdd1e2036f5fe1389d58a27336/crates/oci-distribution/src/client.rs#L823-L835

I think would be better if one of you maintainers could upload the `hello-wasm` image to Google Artifacts. Then I'll add the test 🤔

**Comment by thomastaylor312** [comment]
@bacongobbler I'm not sure if we can easily test this due to the need for a google artifacts registry, unless there is one sitting around someone will let us use. However, that would also make us dependent on whoever owns that registry

**Comment by bacongobbler** [comment]
I figured there should be a way to set up a mock authentication server where we could introspect the request data, but that seems out of scope for this PR.

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Thanks for the fix on this @jiayihu!

File: crates/oci-distribution/src/client.rs
Comment (bacongobbler) [inline crates/oci-distribution/src/client.rs:272]: In the future, a more idiomatic approach would be:

```
if let Some(s) = service {
    query.push("service", s);
}
```

Using `if let` removes the extra `unwrap` call and cleans up the code to be more readable.

https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html

Comment (thomastaylor312) [reply crates/oci-distribution/src/client.rs:272]: Haha @bacongobbler I caught that after I merged and already did it locally. PR coming soon

Comment (jiayihu) [reply crates/oci-distribution/src/client.rs:272]: You're right @bacongobbler, I had a feeling it felt wrong but clippy wasn't saying anything. Sorry 😅

**Pull request merged**
